p2p/src/net/default_backend/types.rs

```diff
@@ -73,6 +73,10 @@ impl P2pTimestamp {
     pub fn as_duration_since_epoch(&self) -> Duration {
         Duration::from_secs(self.0)
     }
+
+    pub fn from_time(time: Time) -> Self {
+        Self::from_duration_since_epoch(time.as_duration_since_epoch())
+    }
```

Comment:
Same as for `BlockTimestamp`, should we have `impl From<Time> for P2pTimestamp` instead of/in addition to this?

Reply:
Same issue